lib/app/utils/hl7_validator.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Inferno
+  # A validator that calls out to the HL7 validator API
+  class HL7Validator
+    @validator_url = nil
+
+    def initialize(validator_url)
+      @validator_url = validator_url
+    end
+
+    def validate(resource, fhir_models_klass, profile = nil)
+      profile ||= fhir_models_klass::Definitions.resource_definition(resource.resourceType).url
+      result = RestClient.post "#{@validator_url}/validate", resource.to_json, params: { profile: profile }
+      outcome = fhir_models_klass.from_contents(result.body)
```

Comment:
`fhir_models_klass` is coming through as a string, at least in some cases:
![Screen Shot 2019-12-16 at 8 39 30 AM](https://user-images.githubusercontent.com/15969967/70911311-a420c300-1fdf-11ea-9914-34f9d91ef717.png)
I think `from_contents` is usually used on the base FHIR class, like `FHIR.from_contents`, as a way to instantiate a FHIR model without having to specify what resource type it is in advance.

Reply:
That was, in fact, the intention of that variable. Seems like sometimes that may not be what's being passed to it though.

Reply:
@Jammjammjamm If you get a chance, can you verify that you can no longer reproduce this issue?

Reply:
The issue has changed:
![Screen Shot 2019-12-18 at 12 22 23 PM](https://user-images.githubusercontent.com/15969967/71108449-1d134c80-2191-11ea-9139-de4adfdeb8c8.png)
It should be `FHIR.from_contents` or `FHIR::DSTU2.from_contents`, not `FHIR::SomeResourceType.from_contents`. Or if you want to instantiate that specific class you should be able to use something like`klass.new(JSON.parse(whatever))`.

Reply:
This issue should be fixed. The only note I have is that the validation will still be a touch off with DSTU2 resources (though it won't error out), because the FHIR Validator service that we've built only supports R4 at the moment.